### PR TITLE
work with CVS statistic

### PR DESCRIPTION
**Pull request by kromch**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+import java.io.IOException;
+import java.util.HashMap;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        String line;
+        HashMap<String, Integer> st = new HashMap<>();
+        try (BufferedReader bri = new BufferedReader(new FileReader(fromFileName));
+                BufferedWriter bro = new BufferedWriter(new FileWriter(toFileName))) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:15]: “write to file” or “create report” - are a separate operations. So you should create separate methods for them.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+            while ((line = bri.readLine()) != null) {
+                String[] em = line.split(",");
+                if (em.length == 2) {
+                    if (st.containsKey(em[0].trim())) {
+                        st.replace(em[0].trim(), st.get(em[0].trim()) + Integer.parseInt(em[1]
+                                .trim()));
+                    } else {
+                        st.put(em[0].trim(), Integer.parseInt(em[1].trim()));
+                    }
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:24]: You call em[0].trim() 4 times and em[1].trim() 2 times. Better to introduce it as local variable.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,47 @@
...
+                    } else {
+                        dif -= st.get(key);
+                    }
+                    i++;
+                }
+                bro.write(tmp);
+                bro.write(String.format("result,%d\n", Math.abs(dif)));
+            }
+        } catch (IOException | NumberFormatException e) {
+            e.printStackTrace();
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:44]: Stack trace is printed by default, it is bad practice. Please, re-throw RuntimeException with original exception and some message in the parameters.

**Review by petro-ushchuk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,59 @@
...
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        String line;
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:12]: Better move this variable inside `try` block

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,59 @@
...
+        HashMap<String, Integer> st = new HashMap<>();
+        try (BufferedReader bri = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = bri.readLine()) != null) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:15]: Introduce `String line = bri.readLine()` before `while` block

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,59 @@
...
+                    }
+                }
+            }
+        } catch (IOException | NumberFormatException e) {
+            if (e instanceof IOException) {
+                throw new RuntimeException(e.getMessage() + "Problem with input file");
+            } else {
+                throw new RuntimeException(e.getMessage());
+            }
+        }
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:33]: Make 2 `catch` blocks instead

Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:33]: Divide program logic

Large methods is very difficult to read and understand. So methods should be short, between 5-15 lines. And it’s important for you to remember that one method should be responsible for one task.

read and write - are a separate operations. So we should create separate methods for them.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,7 +1,59 @@
...
+                bro.write(tmp);
+                bro.write(String.format("result,%d\n", Math.abs(dif)));
+            }
+        } catch (IOException | NumberFormatException e) {
+            if (e instanceof IOException) {
+                throw new RuntimeException(e.getMessage() + "Problem with output file");
+            } else {
+                throw new RuntimeException(e.getMessage());
+            }
+        }
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:57]: Make 2 `catch` blocks instead

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 8]
```diff
@@ -1,7 +1,59 @@
...
+import java.util.HashMap;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        String line;
+        HashMap<String, Integer> st = new HashMap<>();
+        try (BufferedReader bri = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = bri.readLine()) != null) {
+                String[] em = line.split(",");
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:16]: `","` should be constant

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 9]
```diff
@@ -1,7 +1,59 @@
...
+                if (em.length == 2) {
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:17]: `2` should be constant

**Review by petro-ushchuk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 10]
```diff
@@ -1,7 +1,59 @@
...
+                for (String key : st.keySet()) {
+                    tmp = String.format("%s,%d\n", key, st.get(key)) + tmp;
+                    if (i == 0) {
+                        dif = st.get(key);
+                    } else {
+                        dif -= st.get(key);
+                    }
+                    i++;
+                }
+                bro.write(tmp);
```
Comment (petro-ushchuk) [inline src/main/java/core/basesyntax/WorkWithFile.java:48]: Use `StringBuilder` instead.

**Review by petro-ushchuk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 5 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: i did better

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 6 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: done

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 10 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: done

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 9 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: done

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 6 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: refactored

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 4 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: i did better

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 7 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: done

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 8 above)
Comment (kromch) [reply src/main/java/core/basesyntax/WorkWithFile.java]: done

**Pull request closed without merging**